misaligned.js

```diff
@@ -1,16 +1,30 @@
 const {expect} = require('chai')
 
-function print_color_map() {
-    const majorColors = ["White", "Red", "Black", "Yellow", "Violet"];
-    const minorColors = ["Blue", "Orange", "Green", "Brown", "Slate"];
+const colorsMap = new Map();
+const majorColors = ["White", "Red", "Black", "Yellow", "Violet"];
+const minorColors = ["Blue", "Orange", "Green", "Brown", "Slate"];
+
+function calculate_color_map(){    
     for (let i = 0; i < majorColors.length; i++) {
-        for (let j = 0; j < minorColors.length; j++) {
-            console.log(`${i * 5 + j} | ${majorColors[i]} | ${minorColors[j]}`);
+        for (let j = 0; j < minorColors.length; j++) {           
+            let pairIndex = i * 5 + j;
+            colorsMap.set(pairIndex, `| ${majorColors[i]} | ${minorColors[j]}`)
         }
     }
-    return majorColors.length * minorColors.length;
+    return colorsMap;
+}
+
+function print_color_map(colorMap) {
+    colorMap.forEach((element, key) => {       
+        console.log(`${key} ${element}`);        
+    });   
 }
 
-result = print_color_map();
-expect(result).equals(25);
-console.log('All is well (maybe!)');
+colormap = calculate_color_map();
+result = print_color_map(colormap);
+
+expect(colormap.size).equals(25);
+expect(colormap.get(0)).equals(undefined);

```

Comment:
Good isolation of forming.
However, the alignment of the separators in each line is not tested. Think of a test for that